### PR TITLE
ci: Add workflow to build official docker images with multiarch

### LINKED ISSUE #80764: Multiarch docker image
### 🚀 The feature, motivation and pitch

Currently, the PyTorch docker image does not support ARM-based CPUs, which may be an issue for some MacBooks with Apple Silicon users. Also, this fact blocks other projects using PyTorch, e.g., https://github.com/mindsdb/mindsdb/issues/2420.

### Alternatives

_No response_

### Additional context

_No response_

cc @malfet

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83437

Resolves https://github.com/pytorch/pytorch/issues/80764

Signed-off-by: Eli Uriegas <seemethere101@gmail.com>

### CODE REVIEW

File: .github/workflows/docker-release.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  # WITH_PUSH: ${{ github.event_name == 'schedule' }}
+
+jobs:
+  build:
+    runs-on: [self-hosted, linux.2xlarge]
+    timeout-minutes: 240
+    strategy:
+      matrix:
+        include:
+          # nvidia specific images don't exist for arm64 so only build the runtime image
```
Comment (ZainRizvi) [inline .github/workflows/docker-release.yml:44]: should this say "only build the runtime and devel images"?

File: docker.Makefile [hunk 2]
```diff
@@ -8,32 +8,54 @@ $(warning WARNING: No docker user found using results from whoami)
...
 # The conda channel to use to install cudatoolkit
 CUDA_CHANNEL              = nvidia
 # The conda channel to use to install pytorch / torchvision
-INSTALL_CHANNEL           = pytorch
+INSTALL_CHANNEL          ?= pytorch
 
-PYTHON_VERSION            = 3.8
-PYTORCH_VERSION           = $(shell git describe --tags --always)
+PYTHON_VERSION           ?= 3.8
+PYTORCH_VERSION          ?= $(shell git describe --tags --always)
```
Comment (ZainRizvi) [inline docker.Makefile:29]: Why make these sets optional? Do we expect all these variables to have a default values set?

**Review by ZainRizvi — APPROVED** [review]

File: .github/workflows/docker-release.yml (hunk 1 above)
Comment (seemethere) [reply .github/workflows/docker-release.yml:38]: Oh this is specific to arm64

File: docker.Makefile (hunk 2 above)
Comment (seemethere) [reply docker.Makefile:29]: Making these optional allows us to have sane defaults and also make it so that we can override them with environment variables within CI

Comment (seemethere) [reply docker.Makefile:29]: Otherwise we'd have to pass them as make variables :(

File: docker.Makefile [hunk 3]
```diff
@@ -1,39 +1,68 @@
...
 BASE_RUNTIME              = ubuntu:18.04
-BASE_DEVEL                = nvidia/cuda:$(CUDA_VERSION)-cudnn$(CUDNN_VERSION)-devel-ubuntu18.04
+ifeq ("$(CUDA_VERSION_FULL)","cpu")
+MUTEX_PACKAGE             = cpuonly
+BASE_DEVEL                = $(BASE_RUNTIME)
+else
+CUDNN_VERSION             = 8
+MUTEX_PACKAGE             = cudatoolkit=$(CUDA_VERSION_MAJOR_MINOR)
+BASE_DEVEL                = nvidia/cuda:$(CUDA_VERSION_FULL)-cudnn$(CUDNN_VERSION)-devel-ubuntu18.04
+endif
```
Comment (malfet) [inline docker.Makefile:21]: Those changes do not seem to be specific to multiarch, mind moving them to a separate PR?

**Review by malfet — APPROVED** [review]

File: .github/workflows/docker-release.yml (hunk 1 above)
Comment (ZainRizvi) [reply .github/workflows/docker-release.yml:38]: Would it be accurate to say "# nvidia specific images don't exist for arm64 so only build the runtime image for them and not the devel image"?

**Review by ZainRizvi — APPROVED** [review]
